### PR TITLE
Clean up profiler

### PR DESCRIPTION
**Pull request by pkowal1982**
- Fix issue where you can't select by mouse all frames (for example frame 0 can be selected only by edit field)
- Disable frame edit field when no data available
- Disable clear button when nothing to clean
- Align cursor to the middle of selected frame (it was misplaced)
- Remove some dead code

### CODE REVIEW

**Comment by Calinou** [comment]
@pkowal1982 Could you look into rebasing this pull request on the latest `master` branch?

**Comment by pkowal1982** [comment]
@Calinou yeah, I'll try to look into it before this weekend.

**Comment by pkowal1982** [comment]
@Calinou Done.

File: editor/debugger/editor_profiler.cpp [hunk 1]
```diff
@@ -43,28 +43,32 @@ void EditorProfiler::_make_metric_ptrs(Metric &m) {
...
+
 void EditorProfiler::add_frame_metric(const Metric &p_metric, bool p_final) {
 	++last_metric;
 	if (last_metric >= frame_metrics.size()) {
 		last_metric = 0;
 	}
 
+	total_metrics++;
+	if (total_metrics > frame_metrics.size())
+		total_metrics = frame_metrics.size();
```
Comment (Faless) [inline editor/debugger/editor_profiler.cpp:58]:
```cpp
	if (total_metrics > frame_metrics.size()) {
		total_metrics = frame_metrics.size();
	}
```

File: editor/debugger/editor_profiler.cpp [hunk 2]
```diff
@@ -43,28 +43,32 @@ void EditorProfiler::_make_metric_ptrs(Metric &m) {
...
-			}
-		}
-	}
+	clear_button->set_disabled(false);
+	cursor_metric_edit->set_editable(true);
+	cursor_metric_edit->set_max(p_metric.frame_number);
+	cursor_metric_edit->set_min(_get_frame_metric(0).frame_number);
+
+	if (!seeking)
+		cursor_metric_edit->set_value(p_metric.frame_number);
```
Comment (Faless) [inline editor/debugger/editor_profiler.cpp:70]:
```cpp
	if (!seeking) {
		cursor_metric_edit->set_value(p_metric.frame_number);
	}
```

**Review by Faless — CHANGES_REQUESTED** [review]
Haven't tried the changes yet, but see comments on style (we always require braces in control flows)

**Review by Faless — APPROVED** [review]
I've tested it a bit, it's a great improvement.
There are still issues when you select a frame and then close the game without resuming it (the play/stop button stays disabled), but that's already happening in master so we can address that in a separate PR.
Great job :trophy: !

**Pull request merged**